### PR TITLE
Adds coverage for duty options

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

HOTT-<TODO>

### What?

I have added/removed/altered:

- [x] Added base spec to cover all base behaviour
- [x] Added concrete specs to validate data-driven changes to expected duty option tables (e.g. ids, locales, categories, footnotes, etc)
- [x] Added factories to make driving of the duty options easier

### Why?

I am doing this because:

- Trying to add a new duty option for excise and finding it a bit too difficult to test - coming into all the evaluators just for some simple presentational changes

### CODE REVIEW

File: app/models/api/measure.rb [hunk 1]
```diff
@@ -13,19 +13,14 @@ class Measure < Api::Base
     meta_attribute :duty_calculator, :source
 
     attributes :id,
-               :additional_code,
```
Comment (willfish) [inline app/models/api/measure.rb:16]: These were creating getters that are handled by the defined relationships in the DSL for an api resource

File: app/models/duty_calculator.rb [hunk 2]
```diff
@@ -14,12 +14,11 @@ def options
       option = {}
       option[:key] = option_klass.id
       option[:evaluation] = option_klass.new(measure, additional_duty_rows, vat_measure).option
-      option[:priority] = option_klass::PRIORITY
```
Comment (willfish) [inline app/models/duty_calculator.rb:17]: This has moved to the evaluation for consistency

File: app/models/duty_calculator.rb [hunk 3]
```diff
@@ -34,7 +33,10 @@ def additional_duty_rows
         next if option_klass.nil?
         next if measure.all_duties_zero?
 
-        acc << option_klass.new(measure, [], nil).option
+        option = {}
+        option[:key] = option_klass.id
+        option[:evaluation] = option_klass.new(measure, [], nil).option
```
Comment (willfish) [inline app/models/duty_calculator.rb:38]: Made the additional options consistent with the other types of duty option. Could move this process to a shared builder at some point (another PR)

File: app/services/duty_options/base.rb [hunk 4]
```diff
@@ -108,7 +112,7 @@ def duty_totals_row
     def vat_evaluation
       @vat_evaluation ||= ExpressionEvaluators::Vat.new(
         vat_measure,
-        vat_measure.component,
+        vat_measure.measure_components.first,
```
Comment (willfish) [inline app/services/duty_options/base.rb:115]: VAT measures only ever have a single measure component expressed as an ad valorem percentage

File: spec/controllers/steps/additional_code_controller_spec.rb [hunk 5]
```diff
@@ -35,7 +35,7 @@
       end
 
       it { expect { response }.to change(user_session, :additional_code_uk).from({}).to('105' => '2300') }
-      it { expect(response).to redirect_to(vat_path) }
+      it { expect(response).to redirect_to(excise_path('306')) }
```
Comment (willfish) [inline spec/controllers/steps/additional_code_controller_spec.rb:38]: I updated the fixture to include an excise applicable additional code. We're testing the routing logic elsewhere. This just validates the redirect happens

File: spec/decorators/confirmation_decorator_spec.rb [hunk 6]
```diff
@@ -65,7 +65,7 @@
         { key: 'certificate_of_origin', label: 'Certificate of origin', value: 'Yes' },
         { key: 'customs_value', label: 'Customs value', value: '£1,200.00' },
         { key: 'measure_amount', label: 'Import quantity', value: '100 x 100 kg' },
-        { key: 'excise', label: 'Excise additional code', value: 'X444, X369' },
+        { key: 'excise', label: 'Excise additional code', value: '444, 369' },
```
Comment (willfish) [inline spec/decorators/confirmation_decorator_spec.rb:68]: These should always be presented using without the type of the additional code

File: spec/factories/api/additional_code.rb [hunk 7]
```diff
@@ -0,0 +1,28 @@
+FactoryBot.define do
+  sequence(:additional_code_sid)
+
+  factory :api_additional_code, class: 'Api::AdditionalCode' do
```
Comment (willfish) [inline spec/factories/api/additional_code.rb:4]: Unfortunately their is a step factory that reserves the :additional_code key

File: spec/factories/duty_option_result.rb [hunk 8]
```diff
@@ -60,6 +59,24 @@
     source { 'uk' }
   end
 
+  #   trait :additional_duty do
```
Comment (willfish) [inline spec/factories/duty_option_result.rb:62]: I need to reinstate this once I've merged this PR and am validating excise

File: spec/services/duty_options/additional_duty/excise_spec.rb [hunk 9]
```diff
@@ -0,0 +1,22 @@
+# RSpec.describe DutyOptions::AdditionalDuty::Excise, :user_session do
```
Comment (willfish) [inline spec/services/duty_options/additional_duty/excise_spec.rb:1]: I'll uncomment this when I rebase onto my excise duty calculation branch

File: spec/factories/api/definition.rb [hunk 10]
```diff
@@ -0,0 +1,19 @@
+FactoryBot.define do
+  factory :definition, class: 'Api::Definition' do
```
Comment (jebw) [inline spec/factories/api/definition.rb:2]: Is it worth namespacing all the factories for consistency - ie, `api_definition` ? Would avoid potential future name classes with models as well?

**Review by jebw — APPROVED** [review]
Simplifies the spec code a lot - nice!

File: spec/factories/api/definition.rb (hunk 10 above)
Comment (willfish) [reply spec/factories/api/definition.rb:2]: I reckon it might. Let's see if we collide in future and I'll migrate them then.

**Pull request merged**